diff --git a/DifferenceBuffer/AssemblyInfo.cs b/DifferenceBuffer/AssemblyInfo.cs
index 74afe2a..0264812 100644
--- a/DifferenceBuffer/AssemblyInfo.cs
+++ b/DifferenceBuffer/AssemblyInfo.cs
@@ -7,6 +7,7 @@ using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
 using System.Runtime.ConstrainedExecution;
 using System.Security.Permissions;
+using Microsoft.VisualStudio.ExtensibilityHosting;
 
 //
 // General Information about an assembly is controlled through the following 
@@ -18,4 +19,5 @@ using System.Security.Permissions;
 [assembly: AssemblyTrademark ("")]
 [assembly: AssemblyCulture ("")]
 [assembly: ReliabilityContract(Consistency.MayCorruptProcess, Cer.MayFail)]
-[assembly: System.ComponentModel.Composition.AllowNonPublicComposition()]
+[assembly: System.ComponentModel.Composition.AllowNonPublicComposition]
+[assembly: VsCatalogName("Microsoft.VisualStudio.Editor")]
diff --git a/DifferenceBuffer/DifferenceBuffer.csproj b/DifferenceBuffer/DifferenceBuffer.csproj
index 43aca21..4b5fa69 100644
--- a/DifferenceBuffer/DifferenceBuffer.csproj
+++ b/DifferenceBuffer/DifferenceBuffer.csproj
@@ -39,6 +39,10 @@
       <SpecificVersion>False</SpecificVersion>
       <HintPath>C:\Program Files\Microsoft Visual Studio 10.0\Common7\IDE\CommonExtensions\Platform\Editor\Microsoft.VisualStudio.Editor.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.VisualStudio.ExtensibilityHosting, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <RequiredTargetFramework>4.0</RequiredTargetFramework>
+      <Private>False</Private>
+    </Reference>
     <Reference Include="Microsoft.VisualStudio.Text.Data, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
       <HintPath>C:\Program Files\Microsoft Visual Studio 10.0\Common7\IDE\CommonExtensions\Platform\Editor\Microsoft.VisualStudio.Text.Data.dll</HintPath>
diff --git a/VSDelta/Properties/AssemblyInfo.cs b/VSDelta/Properties/AssemblyInfo.cs
index 8c01d7d..3fcedd4 100644
--- a/VSDelta/Properties/AssemblyInfo.cs
+++ b/VSDelta/Properties/AssemblyInfo.cs
@@ -1,6 +1,7 @@
 ﻿using System.Reflection;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
+using Microsoft.VisualStudio.ExtensibilityHosting;
 
 // General Information about an assembly is controlled through the following 
 // set of attributes. Change these attribute values to modify the information
@@ -35,4 +36,5 @@ using System.Runtime.InteropServices;
 [assembly: AssemblyVersion("1.0.0.0")]
 [assembly: AssemblyFileVersion("1.0.0.0")]
 
-[assembly: System.ComponentModel.Composition.AllowNonPublicComposition]
\ No newline at end of file
+[assembly: System.ComponentModel.Composition.AllowNonPublicComposition]
+[assembly: VsCatalogName("Microsoft.VisualStudio.Editor")]
diff --git a/VSDelta/VSDelta-screenshot.jpg b/VSDelta/VSDelta-screenshot.jpg
new file mode 100644
index 0000000..53ddc58
Binary files /dev/null and b/VSDelta/VSDelta-screenshot.jpg differ
diff --git a/VSDelta/VSDelta-screenshot.png b/VSDelta/VSDelta-screenshot.png
new file mode 100644
index 0000000..2f2148b
Binary files /dev/null and b/VSDelta/VSDelta-screenshot.png differ
diff --git a/VSDelta/VSDelta.csproj b/VSDelta/VSDelta.csproj
index f753a21..a05661f 100644
--- a/VSDelta/VSDelta.csproj
+++ b/VSDelta/VSDelta.csproj
@@ -17,7 +17,7 @@
     <DebugSymbols>true</DebugSymbols>
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
-    <OutputPath>C:\Program Files\Microsoft Visual Studio 10.0\Common7\IDE\Components\</OutputPath>
+    <OutputPath>C:\Program Files\Microsoft Visual Studio 10.0\Common7\IDE\Extensions\VSDelta\</OutputPath>
     <DefineConstants>DEBUG;TRACE</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
@@ -34,26 +34,36 @@
     <Reference Include="Microsoft.VisualStudio.CoreUtility, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
       <HintPath>C:\Program Files\Microsoft Visual Studio 10.0\Common7\IDE\CommonExtensions\Platform\Editor\Microsoft.VisualStudio.CoreUtility.dll</HintPath>
+      <Private>False</Private>
     </Reference>
     <Reference Include="Microsoft.VisualStudio.Editor, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
       <HintPath>C:\Program Files\Microsoft Visual Studio 10.0\Common7\IDE\CommonExtensions\Platform\Editor\Microsoft.VisualStudio.Editor.dll</HintPath>
+      <Private>False</Private>
+    </Reference>
+    <Reference Include="Microsoft.VisualStudio.ExtensibilityHosting, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <RequiredTargetFramework>4.0</RequiredTargetFramework>
+      <Private>False</Private>
     </Reference>
     <Reference Include="Microsoft.VisualStudio.Text.Data, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
       <HintPath>C:\Program Files\Microsoft Visual Studio 10.0\Common7\IDE\CommonExtensions\Platform\Editor\Microsoft.VisualStudio.Text.Data.dll</HintPath>
+      <Private>False</Private>
     </Reference>
     <Reference Include="Microsoft.VisualStudio.Text.Logic, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
       <HintPath>C:\Program Files\Microsoft Visual Studio 10.0\Common7\IDE\CommonExtensions\Platform\Editor\Microsoft.VisualStudio.Text.Logic.dll</HintPath>
+      <Private>False</Private>
     </Reference>
     <Reference Include="Microsoft.VisualStudio.Text.UI, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
       <HintPath>C:\Program Files\Microsoft Visual Studio 10.0\Common7\IDE\CommonExtensions\Platform\Editor\Microsoft.VisualStudio.Text.UI.dll</HintPath>
+      <Private>False</Private>
     </Reference>
     <Reference Include="Microsoft.VisualStudio.Text.UI.Wpf, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
       <HintPath>C:\Program Files\Microsoft Visual Studio 10.0\Common7\IDE\CommonExtensions\Platform\Editor\Microsoft.VisualStudio.Text.UI.Wpf.dll</HintPath>
+      <Private>False</Private>
     </Reference>
     <Reference Include="System" />
     <Reference Include="System.ComponentModel.Composition">
@@ -84,6 +94,19 @@
       <Name>DifferenceBuffer</Name>
     </ProjectReference>
   </ItemGroup>
+  <ItemGroup>
+    <None Include="extension.vsixmanifest">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
+  </ItemGroup>
+  <ItemGroup>
+    <Content Include="VSDelta-screenshot.jpg">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
+    <Content Include="VSDelta-screenshot.png">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
+  </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
   <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
        Other similar extension points exist, see Microsoft.Common.targets.
diff --git a/VSDelta/extension.vsixmanifest b/VSDelta/extension.vsixmanifest
new file mode 100644
index 0000000..03d6cb8
--- /dev/null
+++ b/VSDelta/extension.vsixmanifest
@@ -0,0 +1,31 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<VSIX xmlns="http://schemas.microsoft.com/developer/vsx-schema/2010">
+
+  <Identifier ID="VSDelta">
+    <Name>Delta</Name>
+    <Author>Noah Richards</Author>
+    <Version>0.1</Version>
+    <!-- Could this please collapse whitespace so I don't have to 
+         write it all on one line?!
+      -->
+    <Description>An editor extension that allows the viewing of differences directly inside the normal code editor.  Regular language integration (like completion, parameter help, coloring, quick info, squiggles, etc.) works for the matching and added regions, so you can turn on differencing and continue working.</Description>
+    <Locale>1033</Locale>
+    <!-- If this is added, the extension doesn't load at all.  Awesome.
+    <PreviewImage>VSDelta-screenshot.png</PreviewImage>
+     -->
+    <SupportedVSEdition version="10.0">
+      <Edition>VST_All</Edition>
+      <Edition>Express_All</Edition>
+    </SupportedVSEdition>
+    <SupportedFrameworkRuntimeEdition minversion="4.0" maxversion="4.0" />
+    <SystemComponent>false</SystemComponent>
+  </Identifier>
+
+  <References />
+
+  <Content>
+    <MEFComponent>VSDelta.dll</MEFComponent>
+    <MEFComponent>DifferenceBuffer.dll</MEFComponent>
+  </Content>
+
+</VSIX>
\ No newline at end of file
